graylog2-web-interface/src/components/common/EntityFilters/FilterConfiguration/DateRangeForm.tsx

```diff
@@ -0,0 +1,211 @@
+  align-items: center;
+  justify-content: space-between;
+  margin-bottom: 3px;
+`;
+
+const StyledLabel = styled.label`
+  margin: 0;
+`;
+
+const StyledCheckbox = styled(Checkbox)`
+  &.checkbox {
+    margin: 0;
+  }
+`;
+
+const DateTimeFormat = styled.code`
+  padding: 0;
+`;
+
+const ErrorMessage = styled.span(({ theme }) => css`
+  color: ${theme.colors.variant.dark.danger};
+  font-size: ${theme.fonts.size.small};
+  font-style: italic;
+  padding: 3px 3px 9px;
+  height: 1.5em;
+`);
+
+const DateConfiguration = ({ name: fieldName, label, allTimeLabel }: {
+  name: string,
+  label: string,
+  allTimeLabel: string
+}) => {
+  const { formatTime } = useUserDateTime();
+
+  return (
+    <Field name={fieldName}>
+      {({ field: { value, onChange, name }, meta: { error } }) => {
+        const _onChange = (newValue: string) => onChange({ target: { name, value: newValue } });
+        const onChangeAllTime = () => _onChange(value ? undefined : formatTime(new Date(), 'complete'));
+
+        return (
+          <div>
+            <SectionHeader>
+              <StyledLabel htmlFor={`date-input-${name}`}>{label}</StyledLabel>
+              <StyledCheckbox onChange={onChangeAllTime} checked={!value}>{allTimeLabel}</StyledCheckbox>
+            </SectionHeader>
+            <AbsoluteDateInput name="from"
+                               value={value}
+                               disabled={!value}
+                               onChange={_onChange} />
+            {error && <ErrorMessage>{error}</ErrorMessage>}
+          </div>
+        );
+      }}
+    </Field>
+  );
+};
+
+const useInitialValues = (filter: ValueFilter | undefined) => {
+  const { formatTime } = useUserDateTime();
+
+  if (filter) {
+    const [from, until] = filter.value.split(DATE_SEPARATOR);
+
+    return ({
+      from: from ? formatTime(from, 'complete') : undefined,
+      until: until ? formatTime(until, 'complete') : undefined,
+    });
+  }
+
+  return {
+    from: formatTime(moment().subtract(5, 'minutes'), 'complete'),
+    until: undefined,
+  };
+};
+
+const formatError = 'Format must be: YYYY-MM-DD [HH:mm:ss[.SSS]].';
+const rangeError = 'The "Until" date must come after the "From" date.';
+
+const validate = (values: FormValues) => {
+  let errors: {
+      from?: string,
+      until?: string,
+    } = {};
+
+  if (values.from && !isValidDate(values.from)) {
+    errors = { ...errors, from: formatError };
+  }
+
+  if (values.until && !isValidDate(values.until)) {
+    errors = { ...errors, until: formatError };
+  }
+
+  if (values.from >= values.until) {
+    errors = { ...errors, until: rangeError };
+  }
+
+  if (!values.from && !values.until) {
+    errors = { ...errors, from: 'Either "from" or "until" must be an actual date.' };
+  }
+
+  return errors;
+};
+
+type Props = {
+  onSubmit: (filter: { title: string, value: string }) => void,
+  filter: ValueFilter | undefined,
+}
+
+const DateRangeForm = ({ filter, onSubmit }: Props) => {
```

Comment:
<img width="253" alt="Screenshot 2023-03-20 at 19 05 36" src="https://user-images.githubusercontent.com/10741557/226430288-f88843a7-e2cc-4398-93dc-953153c825d2.png">
We have an error when checking both checkboxes

Reply:
The error is actually intended. Imo it makes no sense to filter from "all time" until "now", especially since date filters can be combined. In this case users can just remove the / all date filters.